### DISCUSSION TITLE
GetUserNewsResourcesUseCase  in Architecture Journey does  not exist in project repository

### DISCUSSION THREAD

**Discussion by btcdevops-apps** [android/nowinandroid#1186] [General]
Hello,

I've gone through the nia codebase and I don't think **GetUserNewsResourcesUseCase** class in Step 2 of the architecture journey ([nowinandroid](https://github.com/android/nowinandroid/tree/main)/[docs](https://github.com/android/nowinandroid/tree/main/docs)
/ArchitectureLearningJourney.md) exists.

Does this document need to be updated or the class is missing in the codebase?

**Comment by padrecedano** [comment]
[In this commit](https://github.com/android/nowinandroid/commit/bd450099fbf7f3997ed97ec3f5c8d110eb46fae8) the responsibility for joining the UserData and the
NewsResources was moved to [`UserNewsResourceRepository`](https://github.com/android/nowinandroid/blob/4699fa9d6c1997f58ce5c3f0491d77215c82d5bb/core/data/src/main/kotlin/com/google/samples/apps/nowinandroid/core/data/repository/UserNewsResourceRepository.kt). 

According to the author of the change: *This way, the work can be done once and shared with all consumers in a SharedFlow, rather than having each consumer perform the join itself by invoking the UseCase*.
